devtools/common/src/main/java/io/quarkus/cli/commands/CreateProject.java

```diff
@@ -92,6 +99,10 @@ public boolean doCreateProject(final Map<String, Object> context) throws IOExcep
         context.put(SOURCE_TYPE, sourceType);
         context.put(BUILD_TOOL, buildTool);
 
+        if (extensions != null && extensions.stream().anyMatch(e -> e.toLowerCase().contains("spring-web"))) {
+            context.put(IS_SPRING, Boolean.TRUE);
```

Comment:
add the Spring detection at `doCreateProject` time instead of when setting the `buildTool` so that call order isn't affecting the result.

Reply:
I'm not sure to understand what do you mean. Are you suggesting that the spring detection should be done in the `CreateProjectMojo.execute()` method? If I'm not wrong that would be a contradiction with what @gsmet is saying [here](https://github.com/quarkusio/quarkus/pull/3492#discussion_r313299096), isn't it?

Reply:
@aureamunoz I think I am saying the same as @gsmet.

A consumer may invoke the `buildTool` method before the new `extensions`, it would result in not having the extensions list to check for Spring.
To be sure the order of the calls is not affecting this, I would suggest to move this logic in the `CreateProject. doCreateProject` method which is always called at the end ("resolve the extensions before creating the project").

Reply:
Hi @ia3andy,
This logic is already in the `CreateProject.doCreateProject` after having invoked the `extensions` and the `buildTool` methods.

Reply:
@aureamunoz ok then, sorry about that, Nevermind, I've got confused by GitHub layout showing them like it was in `buildTool` 😅

Reply:
![image](https://user-images.githubusercontent.com/2223984/63330149-9b6ef780-c333-11e9-801a-a35b6c633c3f.png)

Reply:
heheh :+1: , no problem, it reassures me :blush: